identity/app/form/PrivacyMapping.scala

```diff
@@ -90,15 +90,16 @@ case class PrivacyFormData(
       case Some(_) => allowThirdPartyProfiling
     }
 
-    val newConsents = for {
-      oldConsent <- oldUserDO.consents
-      newConsent <- consents
-    } yield {
-      if (oldConsent.id == newConsent.id)
-        newConsent
-      else
-        oldConsent
-    }
+    val newConsents =
+        for {
+          oldConsent <- if (oldUserDO.consents.isEmpty) defaultConsents else oldUserDO.consents
```

Comment:
using a for comprehension for this is kind of confusing to me (and I think the reason this bug came about) - it's more of a `consents.union(...).foldLeft(...)`

would be good to chat about `defaultConsents` as well - I wonder whether our use case has changed since we first put them in